{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "b5f23131",
   "metadata": {},
   "outputs": [],
   "source": [
    "import scanpy as sc\n",
    "import numpy as np\n",
    "import pandas as pd \n",
    "import itertools as it"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "6179c5e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"Supplementary_table_5.csv\",index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "0a268ae9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MYC</th>\n",
       "      <td>GasperiniShendure2019_lowMOI</td>\n",
       "      <td>ReplogleWeissman2022_rpe1</td>\n",
       "      <td>ReplogleWeissman2022_K562_gwps</td>\n",
       "      <td>ReplogleWeissman2022_K562_essential</td>\n",
       "      <td>FrangiehIzar2021_RNA</td>\n",
       "      <td>PierceGreenleaf2021_GM12878</td>\n",
       "      <td>PierceGreenleaf2021_K562</td>\n",
       "      <td>PierceGreenleaf2021_MCF7</td>\n",
       "      <td>FrangiehIzar2021_protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAX</th>\n",
       "      <td>GasperiniShendure2019_highMOI</td>\n",
       "      <td>GasperiniShendure2019_lowMOI</td>\n",
       "      <td>ReplogleWeissman2022_rpe1</td>\n",
       "      <td>ReplogleWeissman2022_K562_gwps</td>\n",
       "      <td>ReplogleWeissman2022_K562_essential</td>\n",
       "      <td>PierceGreenleaf2021_GM12878</td>\n",
       "      <td>PierceGreenleaf2021_K562</td>\n",
       "      <td>PierceGreenleaf2021_MCF7</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CDC5L</th>\n",
       "      <td>TianKampmann2021_CRISPRa</td>\n",
       "      <td>ReplogleWeissman2022_rpe1</td>\n",
       "      <td>ReplogleWeissman2022_K562_gwps</td>\n",
       "      <td>ReplogleWeissman2022_K562_essential</td>\n",
       "      <td>PierceGreenleaf2021_GM12878</td>\n",
       "      <td>PierceGreenleaf2021_K562</td>\n",
       "      <td>PierceGreenleaf2021_MCF7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HSPA5</th>\n",
       "      <td>TianKampmann2021_CRISPRi</td>\n",
       "      <td>ReplogleWeissman2022_rpe1</td>\n",
       "      <td>ReplogleWeissman2022_K562_gwps</td>\n",
       "      <td>ReplogleWeissman2022_K562_essential</td>\n",
       "      <td>PierceGreenleaf2021_GM12878</td>\n",
       "      <td>PierceGreenleaf2021_K562</td>\n",
       "      <td>PierceGreenleaf2021_MCF7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UBC</th>\n",
       "      <td>GasperiniShendure2019_highMOI</td>\n",
       "      <td>GasperiniShendure2019_lowMOI</td>\n",
       "      <td>ReplogleWeissman2022_rpe1</td>\n",
       "      <td>ReplogleWeissman2022_K562_gwps</td>\n",
       "      <td>ReplogleWeissman2022_K562_essential</td>\n",
       "      <td>FrangiehIzar2021_RNA</td>\n",
       "      <td>FrangiehIzar2021_protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PPP2R1A_MAT2A_MVK_ATP5A1</th>\n",
       "      <td>TianKampmann2019_day7neuron</td>\n",
       "      <td>TianKampmann2019_iPSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MVK_PGGT1B_RFK</th>\n",
       "      <td>TianKampmann2019_day7neuron</td>\n",
       "      <td>TianKampmann2019_iPSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HMGCS1_HACD2_UBA1_PPP2R1A</th>\n",
       "      <td>TianKampmann2019_day7neuron</td>\n",
       "      <td>TianKampmann2019_iPSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UBA1_PPP2R1A_UROD_MAT2A_MAP3K12_MMAB</th>\n",
       "      <td>TianKampmann2019_day7neuron</td>\n",
       "      <td>TianKampmann2019_iPSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PPP2R1A_MAT2A_SEPSECS_NDUFA1</th>\n",
       "      <td>TianKampmann2019_day7neuron</td>\n",
       "      <td>TianKampmann2019_iPSC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8402 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                  0  \\\n",
       "MYC                                    GasperiniShendure2019_lowMOI   \n",
       "MAX                                   GasperiniShendure2019_highMOI   \n",
       "CDC5L                                      TianKampmann2021_CRISPRa   \n",
       "HSPA5                                      TianKampmann2021_CRISPRi   \n",
       "UBC                                   GasperiniShendure2019_highMOI   \n",
       "...                                                             ...   \n",
       "PPP2R1A_MAT2A_MVK_ATP5A1                TianKampmann2019_day7neuron   \n",
       "MVK_PGGT1B_RFK                          TianKampmann2019_day7neuron   \n",
       "HMGCS1_HACD2_UBA1_PPP2R1A               TianKampmann2019_day7neuron   \n",
       "UBA1_PPP2R1A_UROD_MAT2A_MAP3K12_MMAB    TianKampmann2019_day7neuron   \n",
       "PPP2R1A_MAT2A_SEPSECS_NDUFA1            TianKampmann2019_day7neuron   \n",
       "\n",
       "                                                                 1  \\\n",
       "MYC                                      ReplogleWeissman2022_rpe1   \n",
       "MAX                                   GasperiniShendure2019_lowMOI   \n",
       "CDC5L                                    ReplogleWeissman2022_rpe1   \n",
       "HSPA5                                    ReplogleWeissman2022_rpe1   \n",
       "UBC                                   GasperiniShendure2019_lowMOI   \n",
       "...                                                            ...   \n",
       "PPP2R1A_MAT2A_MVK_ATP5A1                     TianKampmann2019_iPSC   \n",
       "MVK_PGGT1B_RFK                               TianKampmann2019_iPSC   \n",
       "HMGCS1_HACD2_UBA1_PPP2R1A                    TianKampmann2019_iPSC   \n",
       "UBA1_PPP2R1A_UROD_MAT2A_MAP3K12_MMAB         TianKampmann2019_iPSC   \n",
       "PPP2R1A_MAT2A_SEPSECS_NDUFA1                 TianKampmann2019_iPSC   \n",
       "\n",
       "                                                                   2  \\\n",
       "MYC                                   ReplogleWeissman2022_K562_gwps   \n",
       "MAX                                        ReplogleWeissman2022_rpe1   \n",
       "CDC5L                                 ReplogleWeissman2022_K562_gwps   \n",
       "HSPA5                                 ReplogleWeissman2022_K562_gwps   \n",
       "UBC                                        ReplogleWeissman2022_rpe1   \n",
       "...                                                              ...   \n",
       "PPP2R1A_MAT2A_MVK_ATP5A1                                         NaN   \n",
       "MVK_PGGT1B_RFK                                                   NaN   \n",
       "HMGCS1_HACD2_UBA1_PPP2R1A                                        NaN   \n",
       "UBA1_PPP2R1A_UROD_MAT2A_MAP3K12_MMAB                             NaN   \n",
       "PPP2R1A_MAT2A_SEPSECS_NDUFA1                                     NaN   \n",
       "\n",
       "                                                                        3  \\\n",
       "MYC                                   ReplogleWeissman2022_K562_essential   \n",
       "MAX                                        ReplogleWeissman2022_K562_gwps   \n",
       "CDC5L                                 ReplogleWeissman2022_K562_essential   \n",
       "HSPA5                                 ReplogleWeissman2022_K562_essential   \n",
       "UBC                                        ReplogleWeissman2022_K562_gwps   \n",
       "...                                                                   ...   \n",
       "PPP2R1A_MAT2A_MVK_ATP5A1                                              NaN   \n",
       "MVK_PGGT1B_RFK                                                        NaN   \n",
       "HMGCS1_HACD2_UBA1_PPP2R1A                                             NaN   \n",
       "UBA1_PPP2R1A_UROD_MAT2A_MAP3K12_MMAB                                  NaN   \n",
       "PPP2R1A_MAT2A_SEPSECS_NDUFA1                                          NaN   \n",
       "\n",
       "                                                                        4  \\\n",
       "MYC                                                  FrangiehIzar2021_RNA   \n",
       "MAX                                   ReplogleWeissman2022_K562_essential   \n",
       "CDC5L                                         PierceGreenleaf2021_GM12878   \n",
       "HSPA5                                         PierceGreenleaf2021_GM12878   \n",
       "UBC                                   ReplogleWeissman2022_K562_essential   \n",
       "...                                                                   ...   \n",
       "PPP2R1A_MAT2A_MVK_ATP5A1                                              NaN   \n",
       "MVK_PGGT1B_RFK                                                        NaN   \n",
       "HMGCS1_HACD2_UBA1_PPP2R1A                                             NaN   \n",
       "UBA1_PPP2R1A_UROD_MAT2A_MAP3K12_MMAB                                  NaN   \n",
       "PPP2R1A_MAT2A_SEPSECS_NDUFA1                                          NaN   \n",
       "\n",
       "                                                                5  \\\n",
       "MYC                                   PierceGreenleaf2021_GM12878   \n",
       "MAX                                   PierceGreenleaf2021_GM12878   \n",
       "CDC5L                                    PierceGreenleaf2021_K562   \n",
       "HSPA5                                    PierceGreenleaf2021_K562   \n",
       "UBC                                          FrangiehIzar2021_RNA   \n",
       "...                                                           ...   \n",
       "PPP2R1A_MAT2A_MVK_ATP5A1                                      NaN   \n",
       "MVK_PGGT1B_RFK                                                NaN   \n",
       "HMGCS1_HACD2_UBA1_PPP2R1A                                     NaN   \n",
       "UBA1_PPP2R1A_UROD_MAT2A_MAP3K12_MMAB                          NaN   \n",
       "PPP2R1A_MAT2A_SEPSECS_NDUFA1                                  NaN   \n",
       "\n",
       "                                                             6  \\\n",
       "MYC                                   PierceGreenleaf2021_K562   \n",
       "MAX                                   PierceGreenleaf2021_K562   \n",
       "CDC5L                                 PierceGreenleaf2021_MCF7   \n",
       "HSPA5                                 PierceGreenleaf2021_MCF7   \n",
       "UBC                                   FrangiehIzar2021_protein   \n",
       "...                                                        ...   \n",
       "PPP2R1A_MAT2A_MVK_ATP5A1                                   NaN   \n",
       "MVK_PGGT1B_RFK                                             NaN   \n",
       "HMGCS1_HACD2_UBA1_PPP2R1A                                  NaN   \n",
       "UBA1_PPP2R1A_UROD_MAT2A_MAP3K12_MMAB                       NaN   \n",
       "PPP2R1A_MAT2A_SEPSECS_NDUFA1                               NaN   \n",
       "\n",
       "                                                             7  \\\n",
       "MYC                                   PierceGreenleaf2021_MCF7   \n",
       "MAX                                   PierceGreenleaf2021_MCF7   \n",
       "CDC5L                                                      NaN   \n",
       "HSPA5                                                      NaN   \n",
       "UBC                                                        NaN   \n",
       "...                                                        ...   \n",
       "PPP2R1A_MAT2A_MVK_ATP5A1                                   NaN   \n",
       "MVK_PGGT1B_RFK                                             NaN   \n",
       "HMGCS1_HACD2_UBA1_PPP2R1A                                  NaN   \n",
       "UBA1_PPP2R1A_UROD_MAT2A_MAP3K12_MMAB                       NaN   \n",
       "PPP2R1A_MAT2A_SEPSECS_NDUFA1                               NaN   \n",
       "\n",
       "                                                             8  \n",
       "MYC                                   FrangiehIzar2021_protein  \n",
       "MAX                                                        NaN  \n",
       "CDC5L                                                      NaN  \n",
       "HSPA5                                                      NaN  \n",
       "UBC                                                        NaN  \n",
       "...                                                        ...  \n",
       "PPP2R1A_MAT2A_MVK_ATP5A1                                   NaN  \n",
       "MVK_PGGT1B_RFK                                             NaN  \n",
       "HMGCS1_HACD2_UBA1_PPP2R1A                                  NaN  \n",
       "UBA1_PPP2R1A_UROD_MAT2A_MAP3K12_MMAB                       NaN  \n",
       "PPP2R1A_MAT2A_SEPSECS_NDUFA1                               NaN  \n",
       "\n",
       "[8402 rows x 9 columns]"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "43b15552",
   "metadata": {},
   "outputs": [],
   "source": [
    "info = pd.read_csv(\"scperturb_dataset_info - public_data_merged.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "a3837237",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Index (=FirstauthorLastauthorYear)</th>\n",
       "      <th>dataset_index</th>\n",
       "      <th>Title</th>\n",
       "      <th>doi/url</th>\n",
       "      <th>First Author</th>\n",
       "      <th>Organisms</th>\n",
       "      <th>Modality = Data type</th>\n",
       "      <th>Method</th>\n",
       "      <th>Tissues</th>\n",
       "      <th>Number of perturbations (incl. unperturbed)</th>\n",
       "      <th>...</th>\n",
       "      <th>disease</th>\n",
       "      <th>celltype</th>\n",
       "      <th>tissue type</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>multiple time points?</th>\n",
       "      <th>multiple concentrations?</th>\n",
       "      <th>combined perturbations?</th>\n",
       "      <th>Date of Submission</th>\n",
       "      <th>Year</th>\n",
       "      <th>Mini-Abstract (loosely summarized original Abstract)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AdamsonWeissman2016</td>\n",
       "      <td>GSM2406675_10X001</td>\n",
       "      <td>A multiplexed single-cell CRISPR screening pla...</td>\n",
       "      <td>https://doi.org/10.1016/j.cell.2016.11.048</td>\n",
       "      <td>Britt Adamson</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>RNA</td>\n",
       "      <td>Perturb-seq</td>\n",
       "      <td>K562</td>\n",
       "      <td>9</td>\n",
       "      <td>...</td>\n",
       "      <td>Chronic myeloid leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>01.10.2019</td>\n",
       "      <td>2019</td>\n",
       "      <td>Perturb-seq is applied to dissect the mammalia...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AdamsonWeissman2016</td>\n",
       "      <td>GSM2406677_10X005</td>\n",
       "      <td>A multiplexed single-cell CRISPR screening pla...</td>\n",
       "      <td>https://doi.org/10.1016/j.cell.2016.11.048</td>\n",
       "      <td>Britt Adamson</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>RNA</td>\n",
       "      <td>Perturb-seq</td>\n",
       "      <td>K562</td>\n",
       "      <td>20</td>\n",
       "      <td>...</td>\n",
       "      <td>Chronic myeloid leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>y</td>\n",
       "      <td>01.10.2019</td>\n",
       "      <td>2019</td>\n",
       "      <td>Perturb-seq is applied to dissect the mammalia...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AdamsonWeissman2016</td>\n",
       "      <td>GSM2406681_10X010</td>\n",
       "      <td>A multiplexed single-cell CRISPR screening pla...</td>\n",
       "      <td>https://doi.org/10.1016/j.cell.2016.11.048</td>\n",
       "      <td>Britt Adamson</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>RNA</td>\n",
       "      <td>Perturb-seq</td>\n",
       "      <td>K562</td>\n",
       "      <td>114</td>\n",
       "      <td>...</td>\n",
       "      <td>Chronic myeloid leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>y</td>\n",
       "      <td>01.10.2019</td>\n",
       "      <td>2019</td>\n",
       "      <td>Perturb-seq is applied to dissect the mammalia...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>DixitRegev2016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Perturb-Seq: Dissecting Molecular Circuits wit...</td>\n",
       "      <td>https://doi.org/10.1016/j.cell.2016.11.038</td>\n",
       "      <td>Atray Dixit</td>\n",
       "      <td>Homo sapiens; Mus musculus</td>\n",
       "      <td>RNA</td>\n",
       "      <td>Perturb-seq</td>\n",
       "      <td>K562</td>\n",
       "      <td>31</td>\n",
       "      <td>...</td>\n",
       "      <td>myelogenous leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>NaN</td>\n",
       "      <td>y</td>\n",
       "      <td>partially (TF knockout screen 7 and 13 dpi)</td>\n",
       "      <td>n</td>\n",
       "      <td>y</td>\n",
       "      <td>01.11.2016</td>\n",
       "      <td>2016</td>\n",
       "      <td>This papers introduces Perturb-seq by analyzin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GasperiniShendure2019</td>\n",
       "      <td>atscale</td>\n",
       "      <td>A Genome-wide Framework for Mapping Gene Regul...</td>\n",
       "      <td>https://doi.org/10.1016/j.cell.2018.11.029</td>\n",
       "      <td>Molly Gasperini</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>RNA</td>\n",
       "      <td>CROP-seq</td>\n",
       "      <td>K562</td>\n",
       "      <td>43314 (includes combinations)</td>\n",
       "      <td>...</td>\n",
       "      <td>chronic myelogenous leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>y</td>\n",
       "      <td>01.06.2018</td>\n",
       "      <td>2018</td>\n",
       "      <td>The authors present a multiplex, expression qu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GasperiniShendure2019</td>\n",
       "      <td>highMOI</td>\n",
       "      <td>A Genome-wide Framework for Mapping Gene Regul...</td>\n",
       "      <td>https://doi.org/10.1016/j.cell.2018.11.029</td>\n",
       "      <td>Molly Gasperini</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>RNA</td>\n",
       "      <td>CROP-seq</td>\n",
       "      <td>K562</td>\n",
       "      <td>39087 (includes combinations)</td>\n",
       "      <td>...</td>\n",
       "      <td>chronic myelogenous leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>y</td>\n",
       "      <td>01.06.2018</td>\n",
       "      <td>2018</td>\n",
       "      <td>The authors present a multiplex, expression qu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GasperiniShendure2019</td>\n",
       "      <td>lowMOI</td>\n",
       "      <td>A Genome-wide Framework for Mapping Gene Regul...</td>\n",
       "      <td>https://doi.org/10.1016/j.cell.2018.11.029</td>\n",
       "      <td>Molly Gasperini</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>RNA</td>\n",
       "      <td>CROP-seq</td>\n",
       "      <td>K562</td>\n",
       "      <td>16531 (includes combinations)</td>\n",
       "      <td>...</td>\n",
       "      <td>chronic myelogenous leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>y</td>\n",
       "      <td>01.06.2018</td>\n",
       "      <td>2018</td>\n",
       "      <td>The authors present a multiplex, expression qu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>NormanWeissman2019</td>\n",
       "      <td>filtered</td>\n",
       "      <td>Exploring genetic interaction manifolds constr...</td>\n",
       "      <td>https://doi.org/10.1126/science.aax4438</td>\n",
       "      <td>Thomas Norman</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>RNA</td>\n",
       "      <td>Perturb-seq</td>\n",
       "      <td>K562</td>\n",
       "      <td>237</td>\n",
       "      <td>...</td>\n",
       "      <td>chronic myelogenous leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>y</td>\n",
       "      <td>01.03.2019</td>\n",
       "      <td>2019</td>\n",
       "      <td>Here, the authors present an analytical framew...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>SchraivogelSteinmetz2020</td>\n",
       "      <td>TAP_SCREEN__chromosome_11_screen</td>\n",
       "      <td>Targeted Perturb-seq enables genome-scale gene...</td>\n",
       "      <td>https://doi.org/10.1038/s41592-020-0837-5</td>\n",
       "      <td>Daniel Schraivogel</td>\n",
       "      <td>Homo sapiens; Mus musculus</td>\n",
       "      <td>RNA</td>\n",
       "      <td>TAP-seq</td>\n",
       "      <td>K562</td>\n",
       "      <td>3105</td>\n",
       "      <td>...</td>\n",
       "      <td>chronic myelogenous leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>y</td>\n",
       "      <td>01.08.2019</td>\n",
       "      <td>2019</td>\n",
       "      <td>This paper introduces targeted Perturb-seq (TA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>SchraivogelSteinmetz2020</td>\n",
       "      <td>TAP_SCREEN__chromosome_8_screen</td>\n",
       "      <td>Targeted Perturb-seq enables genome-scale gene...</td>\n",
       "      <td>https://doi.org/10.1038/s41592-020-0837-5</td>\n",
       "      <td>Daniel Schraivogel</td>\n",
       "      <td>Homo sapiens; Mus musculus</td>\n",
       "      <td>RNA</td>\n",
       "      <td>TAP-seq</td>\n",
       "      <td>K562</td>\n",
       "      <td>4115</td>\n",
       "      <td>...</td>\n",
       "      <td>chronic myelogenous leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>y</td>\n",
       "      <td>01.08.2019</td>\n",
       "      <td>2019</td>\n",
       "      <td>This paper introduces targeted Perturb-seq (TA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>XieHon2017</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Multiplexed engineering and analysis of endoge...</td>\n",
       "      <td>https://doi.org/10.1016/j.molcel.2017.03.007</td>\n",
       "      <td>Shiqi Xie</td>\n",
       "      <td>Homo sapiens; Mus musculus</td>\n",
       "      <td>RNA</td>\n",
       "      <td>Mosaic-seq</td>\n",
       "      <td>K562</td>\n",
       "      <td>241</td>\n",
       "      <td>...</td>\n",
       "      <td>chronic myeloid leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>partially (CRISPR)</td>\n",
       "      <td>01.03.2016</td>\n",
       "      <td>2016</td>\n",
       "      <td>Mosaic-seq is introduced as a method to system...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Index (=FirstauthorLastauthorYear)                     dataset_index  \\\n",
       "0                 AdamsonWeissman2016                 GSM2406675_10X001   \n",
       "1                 AdamsonWeissman2016                 GSM2406677_10X005   \n",
       "2                 AdamsonWeissman2016                 GSM2406681_10X010   \n",
       "7                      DixitRegev2016                               NaN   \n",
       "10              GasperiniShendure2019                           atscale   \n",
       "11              GasperiniShendure2019                           highMOI   \n",
       "12              GasperiniShendure2019                            lowMOI   \n",
       "18                 NormanWeissman2019                          filtered   \n",
       "31           SchraivogelSteinmetz2020  TAP_SCREEN__chromosome_11_screen   \n",
       "32           SchraivogelSteinmetz2020   TAP_SCREEN__chromosome_8_screen   \n",
       "42                         XieHon2017                               NaN   \n",
       "\n",
       "                                                Title  \\\n",
       "0   A multiplexed single-cell CRISPR screening pla...   \n",
       "1   A multiplexed single-cell CRISPR screening pla...   \n",
       "2   A multiplexed single-cell CRISPR screening pla...   \n",
       "7   Perturb-Seq: Dissecting Molecular Circuits wit...   \n",
       "10  A Genome-wide Framework for Mapping Gene Regul...   \n",
       "11  A Genome-wide Framework for Mapping Gene Regul...   \n",
       "12  A Genome-wide Framework for Mapping Gene Regul...   \n",
       "18  Exploring genetic interaction manifolds constr...   \n",
       "31  Targeted Perturb-seq enables genome-scale gene...   \n",
       "32  Targeted Perturb-seq enables genome-scale gene...   \n",
       "42  Multiplexed engineering and analysis of endoge...   \n",
       "\n",
       "                                         doi/url        First Author  \\\n",
       "0     https://doi.org/10.1016/j.cell.2016.11.048       Britt Adamson   \n",
       "1     https://doi.org/10.1016/j.cell.2016.11.048       Britt Adamson   \n",
       "2     https://doi.org/10.1016/j.cell.2016.11.048       Britt Adamson   \n",
       "7     https://doi.org/10.1016/j.cell.2016.11.038         Atray Dixit   \n",
       "10    https://doi.org/10.1016/j.cell.2018.11.029     Molly Gasperini   \n",
       "11    https://doi.org/10.1016/j.cell.2018.11.029     Molly Gasperini   \n",
       "12    https://doi.org/10.1016/j.cell.2018.11.029     Molly Gasperini   \n",
       "18       https://doi.org/10.1126/science.aax4438       Thomas Norman   \n",
       "31     https://doi.org/10.1038/s41592-020-0837-5  Daniel Schraivogel   \n",
       "32     https://doi.org/10.1038/s41592-020-0837-5  Daniel Schraivogel   \n",
       "42  https://doi.org/10.1016/j.molcel.2017.03.007           Shiqi Xie   \n",
       "\n",
       "                     Organisms Modality = Data type       Method Tissues  \\\n",
       "0                 Homo sapiens                  RNA  Perturb-seq    K562   \n",
       "1                 Homo sapiens                  RNA  Perturb-seq    K562   \n",
       "2                 Homo sapiens                  RNA  Perturb-seq    K562   \n",
       "7   Homo sapiens; Mus musculus                  RNA  Perturb-seq    K562   \n",
       "10                Homo sapiens                  RNA     CROP-seq    K562   \n",
       "11                Homo sapiens                  RNA     CROP-seq    K562   \n",
       "12                Homo sapiens                  RNA     CROP-seq    K562   \n",
       "18                Homo sapiens                  RNA  Perturb-seq    K562   \n",
       "31  Homo sapiens; Mus musculus                  RNA      TAP-seq    K562   \n",
       "32  Homo sapiens; Mus musculus                  RNA      TAP-seq    K562   \n",
       "42  Homo sapiens; Mus musculus                  RNA   Mosaic-seq    K562   \n",
       "\n",
       "   Number of perturbations (incl. unperturbed)  ...  \\\n",
       "0                                            9  ...   \n",
       "1                                           20  ...   \n",
       "2                                          114  ...   \n",
       "7                                           31  ...   \n",
       "10               43314 (includes combinations)  ...   \n",
       "11               39087 (includes combinations)  ...   \n",
       "12               16531 (includes combinations)  ...   \n",
       "18                                         237  ...   \n",
       "31                                        3105  ...   \n",
       "32                                        4115  ...   \n",
       "42                                         241  ...   \n",
       "\n",
       "                         disease      celltype tissue type Cancer  \\\n",
       "0       Chronic myeloid leukemia  lymphoblasts   cell_line      y   \n",
       "1       Chronic myeloid leukemia  lymphoblasts   cell_line      y   \n",
       "2       Chronic myeloid leukemia  lymphoblasts   cell_line      y   \n",
       "7           myelogenous leukemia  lymphoblasts         NaN      y   \n",
       "10  chronic myelogenous leukemia  lymphoblasts   cell_line      y   \n",
       "11  chronic myelogenous leukemia  lymphoblasts   cell_line      y   \n",
       "12  chronic myelogenous leukemia  lymphoblasts   cell_line      y   \n",
       "18  chronic myelogenous leukemia  lymphoblasts   cell_line      y   \n",
       "31  chronic myelogenous leukemia  lymphoblasts   cell_line      y   \n",
       "32  chronic myelogenous leukemia  lymphoblasts   cell_line      y   \n",
       "42      chronic myeloid leukemia  lymphoblasts   cell_line      n   \n",
       "\n",
       "                          multiple time points? multiple concentrations?  \\\n",
       "0                                             n                        n   \n",
       "1                                             n                        n   \n",
       "2                                             n                        n   \n",
       "7   partially (TF knockout screen 7 and 13 dpi)                        n   \n",
       "10                                            n                        n   \n",
       "11                                            n                        n   \n",
       "12                                            n                        n   \n",
       "18                                            n                        n   \n",
       "31                                            n                        n   \n",
       "32                                            n                        n   \n",
       "42                                            n                        n   \n",
       "\n",
       "   combined perturbations? Date of Submission  Year  \\\n",
       "0                        n         01.10.2019  2019   \n",
       "1                        y         01.10.2019  2019   \n",
       "2                        y         01.10.2019  2019   \n",
       "7                        y         01.11.2016  2016   \n",
       "10                       y         01.06.2018  2018   \n",
       "11                       y         01.06.2018  2018   \n",
       "12                       y         01.06.2018  2018   \n",
       "18                       y         01.03.2019  2019   \n",
       "31                       y         01.08.2019  2019   \n",
       "32                       y         01.08.2019  2019   \n",
       "42      partially (CRISPR)         01.03.2016  2016   \n",
       "\n",
       "    Mini-Abstract (loosely summarized original Abstract)  \n",
       "0   Perturb-seq is applied to dissect the mammalia...     \n",
       "1   Perturb-seq is applied to dissect the mammalia...     \n",
       "2   Perturb-seq is applied to dissect the mammalia...     \n",
       "7   This papers introduces Perturb-seq by analyzin...     \n",
       "10  The authors present a multiplex, expression qu...     \n",
       "11  The authors present a multiplex, expression qu...     \n",
       "12  The authors present a multiplex, expression qu...     \n",
       "18  Here, the authors present an analytical framew...     \n",
       "31  This paper introduces targeted Perturb-seq (TA...     \n",
       "32  This paper introduces targeted Perturb-seq (TA...     \n",
       "42  Mosaic-seq is introduced as a method to system...     \n",
       "\n",
       "[11 rows x 21 columns]"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info[info['Tissues']==\"K562\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "c2ff85b7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Index (=FirstauthorLastauthorYear)</th>\n",
       "      <th>dataset_index</th>\n",
       "      <th>Title</th>\n",
       "      <th>doi/url</th>\n",
       "      <th>First Author</th>\n",
       "      <th>Organisms</th>\n",
       "      <th>Modality = Data type</th>\n",
       "      <th>Method</th>\n",
       "      <th>Tissues</th>\n",
       "      <th>Number of perturbations (incl. unperturbed)</th>\n",
       "      <th>...</th>\n",
       "      <th>disease</th>\n",
       "      <th>celltype</th>\n",
       "      <th>tissue type</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>multiple time points?</th>\n",
       "      <th>multiple concentrations?</th>\n",
       "      <th>combined perturbations?</th>\n",
       "      <th>Date of Submission</th>\n",
       "      <th>Year</th>\n",
       "      <th>Mini-Abstract (loosely summarized original Abstract)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Liscovitch-BrauerSanjana2021</td>\n",
       "      <td>K562_1</td>\n",
       "      <td>Profiling the genetic determinants of chromati...</td>\n",
       "      <td>https://doi.org/10.1038/s41587-021-00902-x</td>\n",
       "      <td>Liscovitch-Brauer, N.</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>CRISPR–sciATAC</td>\n",
       "      <td>Myeloid Leukemia</td>\n",
       "      <td>22</td>\n",
       "      <td>...</td>\n",
       "      <td>chronic myelogenous leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>01.04.2021</td>\n",
       "      <td>2021</td>\n",
       "      <td>The authors describe a CRISPR-based, single-ce...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Liscovitch-BrauerSanjana2021</td>\n",
       "      <td>K562_2</td>\n",
       "      <td>Profiling the genetic determinants of chromati...</td>\n",
       "      <td>https://doi.org/10.1038/s41587-021-00902-x</td>\n",
       "      <td>Liscovitch-Brauer, N.</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>CRISPR–sciATAC</td>\n",
       "      <td>Myeloid Leukemia</td>\n",
       "      <td>84</td>\n",
       "      <td>...</td>\n",
       "      <td>chronic myelogenous leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>01.04.2021</td>\n",
       "      <td>2021</td>\n",
       "      <td>The authors describe a CRISPR-based, single-ce...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>PierceGreenleaf2021</td>\n",
       "      <td>K562</td>\n",
       "      <td>High-throughput single-cell chromatin accessib...</td>\n",
       "      <td>https://doi.org/10.1038/s41467-021-23213-w</td>\n",
       "      <td>Sarah Pierce</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>Spear-ATAC</td>\n",
       "      <td>K562 (leukemia)</td>\n",
       "      <td>41</td>\n",
       "      <td>...</td>\n",
       "      <td>chronic myelogenous leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>01.11.2020</td>\n",
       "      <td>2020</td>\n",
       "      <td>Chromatin accessibility profiling can identify...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>PierceGreenleaf2021</td>\n",
       "      <td>MCF7</td>\n",
       "      <td>High-throughput single-cell chromatin accessib...</td>\n",
       "      <td>https://doi.org/10.1038/s41467-021-23213-w</td>\n",
       "      <td>Sarah Pierce</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>Spear-ATAC</td>\n",
       "      <td>MCF7 (breast cancer)</td>\n",
       "      <td>41</td>\n",
       "      <td>...</td>\n",
       "      <td>breast adenocarcinoma</td>\n",
       "      <td>mammary epithelial cells</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>01.11.2020</td>\n",
       "      <td>2020</td>\n",
       "      <td>Chromatin accessibility profiling can identify...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>PierceGreenleaf2021</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>High-throughput single-cell chromatin accessib...</td>\n",
       "      <td>https://doi.org/10.1038/s41467-021-23213-w</td>\n",
       "      <td>Sarah Pierce</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>Spear-ATAC</td>\n",
       "      <td>GM12878 (lymphoblastoid cells)</td>\n",
       "      <td>41</td>\n",
       "      <td>...</td>\n",
       "      <td>chronic myelogenous leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>01.11.2020</td>\n",
       "      <td>2020</td>\n",
       "      <td>Chromatin accessibility profiling can identify...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Index (=FirstauthorLastauthorYear) dataset_index  \\\n",
       "14       Liscovitch-BrauerSanjana2021        K562_1   \n",
       "15       Liscovitch-BrauerSanjana2021        K562_2   \n",
       "23                PierceGreenleaf2021          K562   \n",
       "24                PierceGreenleaf2021          MCF7   \n",
       "25                PierceGreenleaf2021       GM12878   \n",
       "\n",
       "                                                Title  \\\n",
       "14  Profiling the genetic determinants of chromati...   \n",
       "15  Profiling the genetic determinants of chromati...   \n",
       "23  High-throughput single-cell chromatin accessib...   \n",
       "24  High-throughput single-cell chromatin accessib...   \n",
       "25  High-throughput single-cell chromatin accessib...   \n",
       "\n",
       "                                       doi/url           First Author  \\\n",
       "14  https://doi.org/10.1038/s41587-021-00902-x  Liscovitch-Brauer, N.   \n",
       "15  https://doi.org/10.1038/s41587-021-00902-x  Liscovitch-Brauer, N.   \n",
       "23  https://doi.org/10.1038/s41467-021-23213-w           Sarah Pierce   \n",
       "24  https://doi.org/10.1038/s41467-021-23213-w           Sarah Pierce   \n",
       "25  https://doi.org/10.1038/s41467-021-23213-w           Sarah Pierce   \n",
       "\n",
       "       Organisms Modality = Data type          Method  \\\n",
       "14  Homo sapiens                 ATAC  CRISPR–sciATAC   \n",
       "15  Homo sapiens                 ATAC  CRISPR–sciATAC   \n",
       "23  Homo sapiens                 ATAC      Spear-ATAC   \n",
       "24  Homo sapiens                 ATAC      Spear-ATAC   \n",
       "25  Homo sapiens                 ATAC      Spear-ATAC   \n",
       "\n",
       "                           Tissues  \\\n",
       "14                Myeloid Leukemia   \n",
       "15                Myeloid Leukemia   \n",
       "23                 K562 (leukemia)   \n",
       "24            MCF7 (breast cancer)   \n",
       "25  GM12878 (lymphoblastoid cells)   \n",
       "\n",
       "   Number of perturbations (incl. unperturbed)  ...  \\\n",
       "14                                          22  ...   \n",
       "15                                          84  ...   \n",
       "23                                          41  ...   \n",
       "24                                          41  ...   \n",
       "25                                          41  ...   \n",
       "\n",
       "                         disease                  celltype tissue type Cancer  \\\n",
       "14  chronic myelogenous leukemia              lymphoblasts   cell_line      y   \n",
       "15  chronic myelogenous leukemia              lymphoblasts   cell_line      y   \n",
       "23  chronic myelogenous leukemia              lymphoblasts   cell_line      y   \n",
       "24         breast adenocarcinoma  mammary epithelial cells   cell_line      y   \n",
       "25  chronic myelogenous leukemia              lymphoblasts   cell_line      y   \n",
       "\n",
       "   multiple time points? multiple concentrations? combined perturbations?  \\\n",
       "14                     n                        n                       n   \n",
       "15                     n                        n                       n   \n",
       "23                     n                        n                       n   \n",
       "24                     n                        n                       n   \n",
       "25                     n                        n                       n   \n",
       "\n",
       "   Date of Submission  Year  \\\n",
       "14         01.04.2021  2021   \n",
       "15         01.04.2021  2021   \n",
       "23         01.11.2020  2020   \n",
       "24         01.11.2020  2020   \n",
       "25         01.11.2020  2020   \n",
       "\n",
       "    Mini-Abstract (loosely summarized original Abstract)  \n",
       "14  The authors describe a CRISPR-based, single-ce...     \n",
       "15  The authors describe a CRISPR-based, single-ce...     \n",
       "23  Chromatin accessibility profiling can identify...     \n",
       "24  Chromatin accessibility profiling can identify...     \n",
       "25  Chromatin accessibility profiling can identify...     \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info[info[\"Modality = Data type\"]==\"ATAC\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "2d3d2cb0",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index (=FirstauthorLastauthorYear)                                           Liscovitch-BrauerSanjana2021\n",
       "dataset_index                                                                                      K562_1\n",
       "Title                                                   Profiling the genetic determinants of chromati...\n",
       "doi/url                                                        https://doi.org/10.1038/s41587-021-00902-x\n",
       "First Author                                                                        Liscovitch-Brauer, N.\n",
       "Organisms                                                                                    Homo sapiens\n",
       "Modality = Data type                                                                                 ATAC\n",
       "Method                                                                                     CRISPR–sciATAC\n",
       "Tissues                                                                                  Myeloid Leukemia\n",
       "Number of perturbations (incl. unperturbed)                                                            22\n",
       "Perturbation                                                                                  CRISPR-cas9\n",
       "disease                                                                              myelogenous leukemia\n",
       "celltype                                                                                     lymphoblasts\n",
       "tissue type                                                                                     cell_line\n",
       "Cancer                                                                                                  y\n",
       "multiple time points?                                                                                 NaN\n",
       "multiple concentrations?                                                                              NaN\n",
       "combined perturbations?                                                                                 n\n",
       "Date of Submission                                                                             01.04.2021\n",
       "Year                                                                                                 2021\n",
       "Mini-Abstract (loosely summarized original Abstract)    The authors describe a CRISPR-based, single-ce...\n",
       "Name: 14, dtype: object"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info.iloc[14,:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "546f7326",
   "metadata": {},
   "outputs": [],
   "source": [
    "mini = info[info['Perturbation']==\"CRISPRi\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "6be53a34",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "mini = mini[mini['celltype']==\"lymphoblasts\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "b52c6fe8",
   "metadata": {},
   "outputs": [],
   "source": [
    "LB1 = []\n",
    "LB2 = []\n",
    "PG = []\n",
    "for index, row in df.iterrows(): \n",
    "    if row.str.contains(\"ReplogleWeissman2022_K562_gwps\").any():\n",
    "        if row.str.contains(\"Liscovitch-BrauerSanjana2021_K562_1\").any():\n",
    "            #print(index)\n",
    "            LB1.append(index)\n",
    "        if row.str.contains(\"Liscovitch-BrauerSanjana2021_K562_2\").any():\n",
    "            #print(index)\n",
    "            LB2.append(index)\n",
    "        if row.str.contains(\"PierceGreenleaf2021_K562\").any():\n",
    "            #print(index)\n",
    "            PG.append(index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "0be49e2f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Index (=FirstauthorLastauthorYear)</th>\n",
       "      <th>dataset_index</th>\n",
       "      <th>Title</th>\n",
       "      <th>doi/url</th>\n",
       "      <th>First Author</th>\n",
       "      <th>Organisms</th>\n",
       "      <th>Modality = Data type</th>\n",
       "      <th>Method</th>\n",
       "      <th>Tissues</th>\n",
       "      <th>Number of perturbations (incl. unperturbed)</th>\n",
       "      <th>...</th>\n",
       "      <th>disease</th>\n",
       "      <th>celltype</th>\n",
       "      <th>tissue type</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>multiple time points?</th>\n",
       "      <th>multiple concentrations?</th>\n",
       "      <th>combined perturbations?</th>\n",
       "      <th>Date of Submission</th>\n",
       "      <th>Year</th>\n",
       "      <th>Mini-Abstract (loosely summarized original Abstract)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AdamsonWeissman2016</td>\n",
       "      <td>GSM2406675_10X001</td>\n",
       "      <td>A multiplexed single-cell CRISPR screening pla...</td>\n",
       "      <td>https://doi.org/10.1016/j.cell.2016.11.048</td>\n",
       "      <td>Britt Adamson</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>RNA</td>\n",
       "      <td>Perturb-seq</td>\n",
       "      <td>K562</td>\n",
       "      <td>9</td>\n",
       "      <td>...</td>\n",
       "      <td>Chronic myeloid leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>01.10.2019</td>\n",
       "      <td>2019</td>\n",
       "      <td>Perturb-seq is applied to dissect the mammalia...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AdamsonWeissman2016</td>\n",
       "      <td>GSM2406677_10X005</td>\n",
       "      <td>A multiplexed single-cell CRISPR screening pla...</td>\n",
       "      <td>https://doi.org/10.1016/j.cell.2016.11.048</td>\n",
       "      <td>Britt Adamson</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>RNA</td>\n",
       "      <td>Perturb-seq</td>\n",
       "      <td>K562</td>\n",
       "      <td>20</td>\n",
       "      <td>...</td>\n",
       "      <td>Chronic myeloid leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>y</td>\n",
       "      <td>01.10.2019</td>\n",
       "      <td>2019</td>\n",
       "      <td>Perturb-seq is applied to dissect the mammalia...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AdamsonWeissman2016</td>\n",
       "      <td>GSM2406681_10X010</td>\n",
       "      <td>A multiplexed single-cell CRISPR screening pla...</td>\n",
       "      <td>https://doi.org/10.1016/j.cell.2016.11.048</td>\n",
       "      <td>Britt Adamson</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>RNA</td>\n",
       "      <td>Perturb-seq</td>\n",
       "      <td>K562</td>\n",
       "      <td>114</td>\n",
       "      <td>...</td>\n",
       "      <td>Chronic myeloid leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>y</td>\n",
       "      <td>01.10.2019</td>\n",
       "      <td>2019</td>\n",
       "      <td>Perturb-seq is applied to dissect the mammalia...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GasperiniShendure2019</td>\n",
       "      <td>atscale</td>\n",
       "      <td>A Genome-wide Framework for Mapping Gene Regul...</td>\n",
       "      <td>https://doi.org/10.1016/j.cell.2018.11.029</td>\n",
       "      <td>Molly Gasperini</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>RNA</td>\n",
       "      <td>CROP-seq</td>\n",
       "      <td>K562</td>\n",
       "      <td>43314 (includes combinations)</td>\n",
       "      <td>...</td>\n",
       "      <td>chronic myelogenous leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>y</td>\n",
       "      <td>01.06.2018</td>\n",
       "      <td>2018</td>\n",
       "      <td>The authors present a multiplex, expression qu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GasperiniShendure2019</td>\n",
       "      <td>highMOI</td>\n",
       "      <td>A Genome-wide Framework for Mapping Gene Regul...</td>\n",
       "      <td>https://doi.org/10.1016/j.cell.2018.11.029</td>\n",
       "      <td>Molly Gasperini</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>RNA</td>\n",
       "      <td>CROP-seq</td>\n",
       "      <td>K562</td>\n",
       "      <td>39087 (includes combinations)</td>\n",
       "      <td>...</td>\n",
       "      <td>chronic myelogenous leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>y</td>\n",
       "      <td>01.06.2018</td>\n",
       "      <td>2018</td>\n",
       "      <td>The authors present a multiplex, expression qu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GasperiniShendure2019</td>\n",
       "      <td>lowMOI</td>\n",
       "      <td>A Genome-wide Framework for Mapping Gene Regul...</td>\n",
       "      <td>https://doi.org/10.1016/j.cell.2018.11.029</td>\n",
       "      <td>Molly Gasperini</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>RNA</td>\n",
       "      <td>CROP-seq</td>\n",
       "      <td>K562</td>\n",
       "      <td>16531 (includes combinations)</td>\n",
       "      <td>...</td>\n",
       "      <td>chronic myelogenous leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>y</td>\n",
       "      <td>01.06.2018</td>\n",
       "      <td>2018</td>\n",
       "      <td>The authors present a multiplex, expression qu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>PierceGreenleaf2021</td>\n",
       "      <td>K562</td>\n",
       "      <td>High-throughput single-cell chromatin accessib...</td>\n",
       "      <td>https://doi.org/10.1038/s41467-021-23213-w</td>\n",
       "      <td>Sarah Pierce</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>Spear-ATAC</td>\n",
       "      <td>K562 (leukemia)</td>\n",
       "      <td>41</td>\n",
       "      <td>...</td>\n",
       "      <td>chronic myelogenous leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>01.11.2020</td>\n",
       "      <td>2020</td>\n",
       "      <td>Chromatin accessibility profiling can identify...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>PierceGreenleaf2021</td>\n",
       "      <td>GM12878</td>\n",
       "      <td>High-throughput single-cell chromatin accessib...</td>\n",
       "      <td>https://doi.org/10.1038/s41467-021-23213-w</td>\n",
       "      <td>Sarah Pierce</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>ATAC</td>\n",
       "      <td>Spear-ATAC</td>\n",
       "      <td>GM12878 (lymphoblastoid cells)</td>\n",
       "      <td>41</td>\n",
       "      <td>...</td>\n",
       "      <td>chronic myelogenous leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>01.11.2020</td>\n",
       "      <td>2020</td>\n",
       "      <td>Chromatin accessibility profiling can identify...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ReplogleWeissman2022</td>\n",
       "      <td>K562_essential</td>\n",
       "      <td>Mapping information-rich genotype-phenotype la...</td>\n",
       "      <td>https://doi.org/10.1016/j.cell.2022.05.013</td>\n",
       "      <td>Joseph Replogle</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>RNA</td>\n",
       "      <td>Perturb-seq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2058</td>\n",
       "      <td>...</td>\n",
       "      <td>chronic myeloid leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>01.10.2022</td>\n",
       "      <td>2022</td>\n",
       "      <td>The authors of this study perform genome-scale...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ReplogleWeissman2022</td>\n",
       "      <td>K562_gwps</td>\n",
       "      <td>Mapping information-rich genotype-phenotype la...</td>\n",
       "      <td>https://doi.org/10.1016/j.cell.2022.05.013</td>\n",
       "      <td>Joseph Replogle</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>RNA</td>\n",
       "      <td>Perturb-seq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9867</td>\n",
       "      <td>...</td>\n",
       "      <td>chronic myeloid leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>01.10.2022</td>\n",
       "      <td>2022</td>\n",
       "      <td>The authors of this study perform genome-scale...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>SchraivogelSteinmetz2020</td>\n",
       "      <td>TAP_SCREEN__chromosome_11_screen</td>\n",
       "      <td>Targeted Perturb-seq enables genome-scale gene...</td>\n",
       "      <td>https://doi.org/10.1038/s41592-020-0837-5</td>\n",
       "      <td>Daniel Schraivogel</td>\n",
       "      <td>Homo sapiens; Mus musculus</td>\n",
       "      <td>RNA</td>\n",
       "      <td>TAP-seq</td>\n",
       "      <td>K562</td>\n",
       "      <td>3105</td>\n",
       "      <td>...</td>\n",
       "      <td>chronic myelogenous leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>y</td>\n",
       "      <td>01.08.2019</td>\n",
       "      <td>2019</td>\n",
       "      <td>This paper introduces targeted Perturb-seq (TA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>SchraivogelSteinmetz2020</td>\n",
       "      <td>TAP_SCREEN__chromosome_8_screen</td>\n",
       "      <td>Targeted Perturb-seq enables genome-scale gene...</td>\n",
       "      <td>https://doi.org/10.1038/s41592-020-0837-5</td>\n",
       "      <td>Daniel Schraivogel</td>\n",
       "      <td>Homo sapiens; Mus musculus</td>\n",
       "      <td>RNA</td>\n",
       "      <td>TAP-seq</td>\n",
       "      <td>K562</td>\n",
       "      <td>4115</td>\n",
       "      <td>...</td>\n",
       "      <td>chronic myelogenous leukemia</td>\n",
       "      <td>lymphoblasts</td>\n",
       "      <td>cell_line</td>\n",
       "      <td>y</td>\n",
       "      <td>n</td>\n",
       "      <td>n</td>\n",
       "      <td>y</td>\n",
       "      <td>01.08.2019</td>\n",
       "      <td>2019</td>\n",
       "      <td>This paper introduces targeted Perturb-seq (TA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Index (=FirstauthorLastauthorYear)                     dataset_index  \\\n",
       "0                 AdamsonWeissman2016                 GSM2406675_10X001   \n",
       "1                 AdamsonWeissman2016                 GSM2406677_10X005   \n",
       "2                 AdamsonWeissman2016                 GSM2406681_10X010   \n",
       "10              GasperiniShendure2019                           atscale   \n",
       "11              GasperiniShendure2019                           highMOI   \n",
       "12              GasperiniShendure2019                            lowMOI   \n",
       "23                PierceGreenleaf2021                              K562   \n",
       "25                PierceGreenleaf2021                           GM12878   \n",
       "26               ReplogleWeissman2022                    K562_essential   \n",
       "27               ReplogleWeissman2022                         K562_gwps   \n",
       "31           SchraivogelSteinmetz2020  TAP_SCREEN__chromosome_11_screen   \n",
       "32           SchraivogelSteinmetz2020   TAP_SCREEN__chromosome_8_screen   \n",
       "\n",
       "                                                Title  \\\n",
       "0   A multiplexed single-cell CRISPR screening pla...   \n",
       "1   A multiplexed single-cell CRISPR screening pla...   \n",
       "2   A multiplexed single-cell CRISPR screening pla...   \n",
       "10  A Genome-wide Framework for Mapping Gene Regul...   \n",
       "11  A Genome-wide Framework for Mapping Gene Regul...   \n",
       "12  A Genome-wide Framework for Mapping Gene Regul...   \n",
       "23  High-throughput single-cell chromatin accessib...   \n",
       "25  High-throughput single-cell chromatin accessib...   \n",
       "26  Mapping information-rich genotype-phenotype la...   \n",
       "27  Mapping information-rich genotype-phenotype la...   \n",
       "31  Targeted Perturb-seq enables genome-scale gene...   \n",
       "32  Targeted Perturb-seq enables genome-scale gene...   \n",
       "\n",
       "                                       doi/url        First Author  \\\n",
       "0   https://doi.org/10.1016/j.cell.2016.11.048       Britt Adamson   \n",
       "1   https://doi.org/10.1016/j.cell.2016.11.048       Britt Adamson   \n",
       "2   https://doi.org/10.1016/j.cell.2016.11.048       Britt Adamson   \n",
       "10  https://doi.org/10.1016/j.cell.2018.11.029     Molly Gasperini   \n",
       "11  https://doi.org/10.1016/j.cell.2018.11.029     Molly Gasperini   \n",
       "12  https://doi.org/10.1016/j.cell.2018.11.029     Molly Gasperini   \n",
       "23  https://doi.org/10.1038/s41467-021-23213-w        Sarah Pierce   \n",
       "25  https://doi.org/10.1038/s41467-021-23213-w        Sarah Pierce   \n",
       "26  https://doi.org/10.1016/j.cell.2022.05.013     Joseph Replogle   \n",
       "27  https://doi.org/10.1016/j.cell.2022.05.013     Joseph Replogle   \n",
       "31   https://doi.org/10.1038/s41592-020-0837-5  Daniel Schraivogel   \n",
       "32   https://doi.org/10.1038/s41592-020-0837-5  Daniel Schraivogel   \n",
       "\n",
       "                     Organisms Modality = Data type       Method  \\\n",
       "0                 Homo sapiens                  RNA  Perturb-seq   \n",
       "1                 Homo sapiens                  RNA  Perturb-seq   \n",
       "2                 Homo sapiens                  RNA  Perturb-seq   \n",
       "10                Homo sapiens                  RNA     CROP-seq   \n",
       "11                Homo sapiens                  RNA     CROP-seq   \n",
       "12                Homo sapiens                  RNA     CROP-seq   \n",
       "23                Homo sapiens                 ATAC   Spear-ATAC   \n",
       "25                Homo sapiens                 ATAC   Spear-ATAC   \n",
       "26                Homo sapiens                  RNA  Perturb-seq   \n",
       "27                Homo sapiens                  RNA  Perturb-seq   \n",
       "31  Homo sapiens; Mus musculus                  RNA      TAP-seq   \n",
       "32  Homo sapiens; Mus musculus                  RNA      TAP-seq   \n",
       "\n",
       "                           Tissues  \\\n",
       "0                             K562   \n",
       "1                             K562   \n",
       "2                             K562   \n",
       "10                            K562   \n",
       "11                            K562   \n",
       "12                            K562   \n",
       "23                 K562 (leukemia)   \n",
       "25  GM12878 (lymphoblastoid cells)   \n",
       "26                             NaN   \n",
       "27                             NaN   \n",
       "31                            K562   \n",
       "32                            K562   \n",
       "\n",
       "   Number of perturbations (incl. unperturbed)  ...  \\\n",
       "0                                            9  ...   \n",
       "1                                           20  ...   \n",
       "2                                          114  ...   \n",
       "10               43314 (includes combinations)  ...   \n",
       "11               39087 (includes combinations)  ...   \n",
       "12               16531 (includes combinations)  ...   \n",
       "23                                          41  ...   \n",
       "25                                          41  ...   \n",
       "26                                        2058  ...   \n",
       "27                                        9867  ...   \n",
       "31                                        3105  ...   \n",
       "32                                        4115  ...   \n",
       "\n",
       "                         disease      celltype tissue type Cancer  \\\n",
       "0       Chronic myeloid leukemia  lymphoblasts   cell_line      y   \n",
       "1       Chronic myeloid leukemia  lymphoblasts   cell_line      y   \n",
       "2       Chronic myeloid leukemia  lymphoblasts   cell_line      y   \n",
       "10  chronic myelogenous leukemia  lymphoblasts   cell_line      y   \n",
       "11  chronic myelogenous leukemia  lymphoblasts   cell_line      y   \n",
       "12  chronic myelogenous leukemia  lymphoblasts   cell_line      y   \n",
       "23  chronic myelogenous leukemia  lymphoblasts   cell_line      y   \n",
       "25  chronic myelogenous leukemia  lymphoblasts   cell_line      y   \n",
       "26      chronic myeloid leukemia  lymphoblasts   cell_line      y   \n",
       "27      chronic myeloid leukemia  lymphoblasts   cell_line      y   \n",
       "31  chronic myelogenous leukemia  lymphoblasts   cell_line      y   \n",
       "32  chronic myelogenous leukemia  lymphoblasts   cell_line      y   \n",
       "\n",
       "   multiple time points? multiple concentrations? combined perturbations?  \\\n",
       "0                      n                        n                       n   \n",
       "1                      n                        n                       y   \n",
       "2                      n                        n                       y   \n",
       "10                     n                        n                       y   \n",
       "11                     n                        n                       y   \n",
       "12                     n                        n                       y   \n",
       "23                     n                        n                       n   \n",
       "25                     n                        n                       n   \n",
       "26                     n                        n                       n   \n",
       "27                     n                        n                       n   \n",
       "31                     n                        n                       y   \n",
       "32                     n                        n                       y   \n",
       "\n",
       "   Date of Submission  Year  \\\n",
       "0          01.10.2019  2019   \n",
       "1          01.10.2019  2019   \n",
       "2          01.10.2019  2019   \n",
       "10         01.06.2018  2018   \n",
       "11         01.06.2018  2018   \n",
       "12         01.06.2018  2018   \n",
       "23         01.11.2020  2020   \n",
       "25         01.11.2020  2020   \n",
       "26         01.10.2022  2022   \n",
       "27         01.10.2022  2022   \n",
       "31         01.08.2019  2019   \n",
       "32         01.08.2019  2019   \n",
       "\n",
       "    Mini-Abstract (loosely summarized original Abstract)  \n",
       "0   Perturb-seq is applied to dissect the mammalia...     \n",
       "1   Perturb-seq is applied to dissect the mammalia...     \n",
       "2   Perturb-seq is applied to dissect the mammalia...     \n",
       "10  The authors present a multiplex, expression qu...     \n",
       "11  The authors present a multiplex, expression qu...     \n",
       "12  The authors present a multiplex, expression qu...     \n",
       "23  Chromatin accessibility profiling can identify...     \n",
       "25  Chromatin accessibility profiling can identify...     \n",
       "26  The authors of this study perform genome-scale...     \n",
       "27  The authors of this study perform genome-scale...     \n",
       "31  This paper introduces targeted Perturb-seq (TA...     \n",
       "32  This paper introduces targeted Perturb-seq (TA...     \n",
       "\n",
       "[12 rows x 21 columns]"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mini"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "e9d85408",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "29"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(LB2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "aa244fd6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "23"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(PG)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "7156f820",
   "metadata": {},
   "outputs": [],
   "source": [
    "LB1 = []\n",
    "LB2 = []\n",
    "PG = []\n",
    "for index, row in df.iterrows(): \n",
    "    if row.str.contains(\"ReplogleWeissman2022_K562_essential\").any():\n",
    "        if row.str.contains(\"Liscovitch-BrauerSanjana2021_K562_1\").any():\n",
    "            #print(index)\n",
    "            LB1.append(index)\n",
    "        if row.str.contains(\"Liscovitch-BrauerSanjana2021_K562_2\").any():\n",
    "            #print(index)\n",
    "            LB2.append(index)\n",
    "        if row.str.contains(\"PierceGreenleaf2021_K562\").any():\n",
    "            #print(index)\n",
    "            PG.append(index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "e65c1bf7",
   "metadata": {},
   "outputs": [],
   "source": [
    "### ok let's look at that "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "b9a1ca1f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['SMARCB1', 'CHD4', 'SETD2', 'H3F3A']"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LB1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "0dc079d9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['YY1',\n",
       " 'RUVBL2',\n",
       " 'ACTB',\n",
       " 'RBBP4',\n",
       " 'SMARCE1',\n",
       " 'ACTR8',\n",
       " 'BPTF',\n",
       " 'CHAF1B',\n",
       " 'SRCAP',\n",
       " 'SSRP1',\n",
       " 'EP400',\n",
       " 'DMAP1',\n",
       " 'MYBBP1A',\n",
       " 'ACTR6',\n",
       " 'BRD8',\n",
       " 'SUPT16H',\n",
       " 'DDX21',\n",
       " 'INO80B',\n",
       " 'TRRAP',\n",
       " 'INO80',\n",
       " 'NFRKB',\n",
       " 'MCRS1',\n",
       " 'ING3',\n",
       " 'KAT5',\n",
       " 'ACTL6A',\n",
       " 'SF3B1',\n",
       " 'RCOR1',\n",
       " 'SIN3A',\n",
       " 'YEATS4']"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LB2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "0161df3a",
   "metadata": {},
   "outputs": [],
   "source": [
    "mini = mini[mini[\"dataset_index\"]!=\"GM12878\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "id": "c6964c86",
   "metadata": {},
   "outputs": [],
   "source": [
    "mini = mini[mini[\"Index (=FirstauthorLastauthorYear)\"]!=\"SchraivogelSteinmetz2020\"]\n",
    "mini = mini[mini[\"Index (=FirstauthorLastauthorYear)\"]!=\"AdamsonWeissman2016\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "id": "98369be0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GasperiniShendure2019_atscale\n",
      "GasperiniShendure2019_highMOI\n",
      "GasperiniShendure2019_lowMOI\n",
      "PierceGreenleaf2021_K562\n",
      "ReplogleWeissman2022_K562_essential\n",
      "ReplogleWeissman2022_K562_gwps\n"
     ]
    }
   ],
   "source": [
    "dset_options = mini['Index (=FirstauthorLastauthorYear)'].unique()\n",
    "for ind in dset_options:\n",
    "    micro = mini[mini['Index (=FirstauthorLastauthorYear)']==ind]\n",
    "    for dset in micro['dataset_index']:\n",
    "        print(ind+\"_\"+dset)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "id": "4792bff5",
   "metadata": {},
   "outputs": [],
   "source": [
    "dsets = mini['Index (=FirstauthorLastauthorYear)']+\"_\"+mini['dataset_index']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "id": "fe173aea",
   "metadata": {},
   "outputs": [],
   "source": [
    "dsets = dsets.to_list()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9b376ffa",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "088b8849",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "41658b6e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bf504c9d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "id": "c30d7f01",
   "metadata": {},
   "outputs": [],
   "source": [
    "shared_perts = pd.DataFrame(0, index=dsets, columns=dsets)\n",
    "\n",
    "for index, row in df.iterrows():\n",
    "    #for pert_ind, pert_row in shared_perts.iterrows():\n",
    "    a = row[row.isin(dsets)].to_list()\n",
    "    for pairing in it.product(a, a):\n",
    "        if pairing[0]!=pairing[1]:\n",
    "            shared_perts.loc[pairing] +=1\n",
    "\n",
    "    #\n",
    "    #\n",
    "    #    print(row)\n",
    "    #    \n",
    "    #    \n",
    "    #    print()\n",
    "    #    break\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9bd6a9b6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "id": "490eeabe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GasperiniShendure2019_atscale</th>\n",
       "      <th>GasperiniShendure2019_highMOI</th>\n",
       "      <th>GasperiniShendure2019_lowMOI</th>\n",
       "      <th>PierceGreenleaf2021_K562</th>\n",
       "      <th>ReplogleWeissman2022_K562_essential</th>\n",
       "      <th>ReplogleWeissman2022_K562_gwps</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GasperiniShendure2019_atscale</th>\n",
       "      <td>0</td>\n",
       "      <td>32</td>\n",
       "      <td>99</td>\n",
       "      <td>0</td>\n",
       "      <td>28</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GasperiniShendure2019_highMOI</th>\n",
       "      <td>32</td>\n",
       "      <td>0</td>\n",
       "      <td>350</td>\n",
       "      <td>2</td>\n",
       "      <td>26</td>\n",
       "      <td>109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GasperiniShendure2019_lowMOI</th>\n",
       "      <td>99</td>\n",
       "      <td>350</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>127</td>\n",
       "      <td>370</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PierceGreenleaf2021_K562</th>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>23</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ReplogleWeissman2022_K562_essential</th>\n",
       "      <td>28</td>\n",
       "      <td>26</td>\n",
       "      <td>127</td>\n",
       "      <td>23</td>\n",
       "      <td>0</td>\n",
       "      <td>2053</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ReplogleWeissman2022_K562_gwps</th>\n",
       "      <td>91</td>\n",
       "      <td>109</td>\n",
       "      <td>370</td>\n",
       "      <td>39</td>\n",
       "      <td>2053</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     GasperiniShendure2019_atscale  \\\n",
       "GasperiniShendure2019_atscale                                    0   \n",
       "GasperiniShendure2019_highMOI                                   32   \n",
       "GasperiniShendure2019_lowMOI                                    99   \n",
       "PierceGreenleaf2021_K562                                         0   \n",
       "ReplogleWeissman2022_K562_essential                             28   \n",
       "ReplogleWeissman2022_K562_gwps                                  91   \n",
       "\n",
       "                                     GasperiniShendure2019_highMOI  \\\n",
       "GasperiniShendure2019_atscale                                   32   \n",
       "GasperiniShendure2019_highMOI                                    0   \n",
       "GasperiniShendure2019_lowMOI                                   350   \n",
       "PierceGreenleaf2021_K562                                         2   \n",
       "ReplogleWeissman2022_K562_essential                             26   \n",
       "ReplogleWeissman2022_K562_gwps                                 109   \n",
       "\n",
       "                                     GasperiniShendure2019_lowMOI  \\\n",
       "GasperiniShendure2019_atscale                                  99   \n",
       "GasperiniShendure2019_highMOI                                 350   \n",
       "GasperiniShendure2019_lowMOI                                    0   \n",
       "PierceGreenleaf2021_K562                                        3   \n",
       "ReplogleWeissman2022_K562_essential                           127   \n",
       "ReplogleWeissman2022_K562_gwps                                370   \n",
       "\n",
       "                                     PierceGreenleaf2021_K562  \\\n",
       "GasperiniShendure2019_atscale                               0   \n",
       "GasperiniShendure2019_highMOI                               2   \n",
       "GasperiniShendure2019_lowMOI                                3   \n",
       "PierceGreenleaf2021_K562                                    0   \n",
       "ReplogleWeissman2022_K562_essential                        23   \n",
       "ReplogleWeissman2022_K562_gwps                             39   \n",
       "\n",
       "                                     ReplogleWeissman2022_K562_essential  \\\n",
       "GasperiniShendure2019_atscale                                         28   \n",
       "GasperiniShendure2019_highMOI                                         26   \n",
       "GasperiniShendure2019_lowMOI                                         127   \n",
       "PierceGreenleaf2021_K562                                              23   \n",
       "ReplogleWeissman2022_K562_essential                                    0   \n",
       "ReplogleWeissman2022_K562_gwps                                      2053   \n",
       "\n",
       "                                     ReplogleWeissman2022_K562_gwps  \n",
       "GasperiniShendure2019_atscale                                    91  \n",
       "GasperiniShendure2019_highMOI                                   109  \n",
       "GasperiniShendure2019_lowMOI                                    370  \n",
       "PierceGreenleaf2021_K562                                         39  \n",
       "ReplogleWeissman2022_K562_essential                            2053  \n",
       "ReplogleWeissman2022_K562_gwps                                    0  "
      ]
     },
     "execution_count": 173,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "shared_perts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dccd86b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ok so that leaves us with a decent number of overlap pairs"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:pertpy5]",
   "language": "python",
   "name": "conda-env-pertpy5-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
