<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8">
        <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="description" content="scPerturb">
        <meta name="keywords" content="scPerturb">
        <title>scPerturb: Single Cell Perturbation Datasets</title>
        <!-- View example and source HTML here: https://picocss.com/examples/classless/ -->
        <link rel="stylesheet" href="pico.classless.min.css">
        <!-- Google tag (gtag.js) -->
        <script async src="https://www.googletagmanager.com/gtag/js?id=G-KTLD8MLVYP"></script>
        <script>
          window.dataLayer = window.dataLayer || [];
          function gtag(){dataLayer.push(arguments);}
          gtag('js', new Date());

          gtag('config', 'G-KTLD8MLVYP');
        </script>
    </head>
    <body>
        <main>
            <figure>
                <img src="scperturb_logo.png" alt="scPerturb logo">
            </figure>
            <h1>Single Cell Perturbation Datasets</h1>
            <p>scPerturb.org is a resource of standardized datasets reporting targeted perturbations with single-cell readouts and to facilitate the development and benchmarking of computational approaches in system biology.</p>
            <h2>Read more and cite scPerturb</h2>
            <p>scPerturb: harmonized single-cell perturbation data
Stefan Peidli, Tessa Durakis Green, Ciyue Shen, Torsten Gross, Joseph Min, Samuele Garda, Bo Yuan, Linus J. Schumacher, Jake P. Taylor-King, Debora S. Marks, Augustin Luna, Nils Blüthgen, Chris Sander
Nature Methods 2024; doi: <a href="https://doi.org/10.1038/s41592-023-02144-y" target="_blank">10.1038/s41592-023-02144-y</a></p>
            <h2>Data Explorer</h2>
            <p>Explore properties of some datasets (does not include all datasets in most recent Zenodo version)<a href="datavzrd/scPerturb_vzrd_v2/dataset_info/index_1.html" target="_blank">here</a> (e.g., links to publications, perturbation type, perturbation counts, etc.)</p>
            <h2>Data</h2>
            <ul>
                <li>RNA-seq datasets: <a href="https://zenodo.org/doi/10.5281/zenodo.7041848" target="_blank">zenodo.org/doi/10.5281/zenodo.7041848</a></li>
                <li>ATAC-seq: <a href="https://zenodo.org/doi/10.5281/zenodo.7058381" target="_blank">zenodo.org/doi/10.5281/zenodo.7058381</a></li>
                <li>Dataset information: <a href="https://www.biorxiv.org/content/biorxiv/early/2022/08/22/2022.08.20.504663/DC1/embed/media-1.xlsx?download=true" target="_blank">bioRxiv Supplementary Table 1</a></li>
            </ul>
            <h2>Source Code</h2>
            <ul>
                <li>Source code and bug reports on <a href="https://github.com/sanderlab/scPerturb" target="_blank">GitHub</a></li>
                <li><a href="https://pypi.org/project/scperturb/" target="_blank">scPerturb Python Package</a> for the use of E-distance to compare and statistically test how much cells in high-dimensional singe-cell data differ (e.g., untreated vs treated)</li>
                <li><a href="https://cran.r-project.org/package=scperturbR" target="_blank">scPerturbR</a>: scperturb for R and Seurat</li>
            </ul>
            <h2>Funding and Acknowledgements</h2>
            <p>National Resource for Network Biology (NRNB, P41GM103504) ● Supported by the Wellcome Leap ∆Tissue Program ● Deutsche Forschungsgemeinschaft (DFG, RTG2424 CompCancer) ● Einstein Stiftung Berlin (Einstein visiting fellow program) ● Computation was in part performed on the HPC for Research cluster of the Berlin Institute of Health</p>
            <a href="https://twitter.com/sandercbio" title="Chris Sander Lab on Twitter" target="_blank">Chris Sander Lab on Twitter: @sandercbio</a>
        </main>
    </body>
</html>
